Index: b/gcc/c-family/c-cppbuiltin.c
===================================================================
--- a/gcc/c-family/c-cppbuiltin.c
+++ b/gcc/c-family/c-cppbuiltin.c
@@ -1408,6 +1408,8 @@ c_cpp_builtins (cpp_reader *pfile)
   TARGET_OS_CPP_BUILTINS ();
   TARGET_OBJFMT_CPP_BUILTINS ();
 
+  builtin_define ("__uefi__");
+
   /* Support the __declspec keyword by turning them into attributes.
      Note that the current way we do this may result in a collision
      with predefined attributes later on.  This can be solved by using
